### PR TITLE
Merge 6.0.x -> 6.3.x

### PR DESCRIPTION
**Pull request by Magisus**
This had conflicts around the project.clj refactors, @justinstoller can you double check those changes please?

### CODE REVIEW

**Comment by Magisus** [comment]
There appears to be an issue with the voom builds after these changes:
```
14:35:12 Deploy project.
14:35:12 + /var/lib/jenkins/bin/lein with-profile user,ci-voom voom deploy
14:35:14 WARNING: record? already refers to: #'clojure.core/record? in namespace: clojure.core.logic, being replaced by: #'clojure.core.logic/record?
14:35:14 Error encountered performing task 'voom' with profile(s): 'user,ci-voom'
14:35:14 Syntax error macroexpanding clojure.core/fn at (clojure/core/logic.clj:1114:5).
14:35:14 Call to clojure.core/fn did not conform to spec.
```
Let's hold off mergeing this till we get that sorted.

File: project.clj [hunk 1]
```diff
@@ -252,9 +212,11 @@
...
   :jvm-opts ["-Djruby.logger.class=com.puppetlabs.jruby_utils.jruby.Slf4jLogger"
                "-XX:+UseG1GC"
-               ~(str "-Xms" (heap-size "1G" "min"))
-               ~(str "-Xmx" (heap-size "2G" "max"))
-               "-XX:+IgnoreUnrecognizedVMOptions"]
+               ~(str "-Xms" (heap-size "1G"))
+               ~(str "-Xmx" (heap-size "2G"))
+               "-XX:+IgnoreUnrecognizedVMOptions"
+               "--add-modules=java.xml.bind"
+               "--add-modules=java.xml.ws"]
```
Comment (justinstoller) [inline project.clj]: Looks like this came with the merge up?

Comment (Magisus) [reply project.clj]: Ah yeah, I guess those weren't part of those changes.

**Comment by Magisus** [comment]
We're fixing the voom issue in the docker image used for the CI jobs. So this can be merged.

**Pull request merged**